_drafts/rspec-style-guide.md

```diff
@@ -0,0 +1,37 @@
+- **Minimize data setup per expectation.** Whether it's in `let` statements or in method calls in a `before` block or the example itself, lots of data setup for each expectation makes it hard to grok. Ideally, you should only need to see the data setup that's necessary to understand the example. Here are a few approaches that can help:
+    - Define default data using a factory system like FactoryGirl. That way, when insantiating classes, you only need to override the fields necessary for that example.
+    - Define objects in the broader context using `let` statements, then override only the fields you need to for the given test.
+    - Create custom matchers that match for something specific you're looking for, so that you don't have to construct complex objects to use for matching.
+- **Always set the pertinent fields on objects in the expectation, even if you know they'll already be set to an appropriate value.** For example, you are specifying what happens when a certain field on an object is set to true. You happen to know that the factory for that object sets that field to true by default. One problem with this is that it leads to test fragility: if you change the default value of that field in the factory, lots of specs could start failing. It also hinders readability: it's not as clear to the reader that that field will be set to true. Go ahead and set it in your example (or a very close `before` block).
+
+## Test Doubles
+
+- **Stub in a before block, mock in the example.** If you're specifying only one behavior per example, there will often be a lot of outgoing messages you *don't* care about for that example, but that still need to work. Stubbing (`allow()`) instead of mocking (`expect()`) is a way to make it clear that you aren't specifying that behavior, just supplying something to get the method call to work. But all those stubs will clutter up your examples, and there will be a lot of duplication. Instead, put all your `allow()` statements in a single `before` block; that sets up the mock collaborators for all the behavior that all the examples will need. Then your examples will only have the `expect()` statements related to what they're specifying. This makes for very clear examples.
+- **But actually spy instead of mock.** When your method calls are stubbed, you don't need to set expectations before calling your production code; the stubs will allow them to pass. Then you're free to use spying instead of mocking. Spies tend to be more readable than mocks, because they follow the "arrange, act, assert" order more clearly.
+
+## Subjects
+
+- **Use the implicit subject when and only when the expectation reads like English.** If `it { is_expected.to... }` makes it harder to understand what you're specifying, and you have to explain it in a surrounding `context` block's name, just use a regular `it` statement instead.
```

Comment:
I want to say that the shorthand, implied-subject, style still works with named subjects. That is if you `subject(:foo) {...}` you can still access `subject` directly in test rather than only by `foo`. Might be wrong though.